### PR TITLE
jscs and jsxcs JS profiling

### PR DESCRIPTION
**Pull request by Pomax**
Clearly needs a better .jscsrc file, but we can now check our .js and .jsx files for style violations

### CODE REVIEW

**Comment by Pomax** [comment]
open question still: any good autoformatters we can use to "help" reformat bad code?

**Comment by thisandagain** [comment]
Awesome. :smile:

File: Gulpfile.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ require('./shared/Gulpfile');
  * files to be written before we move on to the next task,
  * because in this case we can't run parallel tasks.
  */
-gulp.task('default', ['less', 'lint', 'lint-api', 'gallery', 'editor']);
+gulp.task('default', ['less', 'lint', 'jscs', 'lint-api', 'gallery', 'editor']);
```
Comment (Pomax) [inline Gulpfile.js]: api-server needs to be an aggregrate task, too, running both jscs and linting

File: examples/api-server/Gulpfile.js [hunk 2]
```diff
@@ -14,3 +14,15 @@ gulp.task("lint-api", function() {
...
+ * JavaScript style validation, using JSCS
+ */
+gulp.task("jscs-api", function() {
+  var jsxcs = require("gulp-jsxcs");
+  return gulp.src("**/*.js*")
+    .pipe(jsxcs())
+    .pipe(process.stdout);
+});
+
+
```
Comment (Pomax) [inline examples/api-server/Gulpfile.js]: add aggregate task `api-server` much like editor/gallery to capture both lint and jscs

**Pull request merged**